dames.c

```diff
@@ -71,66 +71,164 @@ char pion_init(int ligne, int colonne){
     return pion;
 }
 
+int changement_dans_damier(char tab[][COLONNE_DAMIER],char tab2[][COLONNE_DAMIER]){
 
-
+    for (int ligne = 0; ligne < LIGNE_DAMIER ; ligne++){
+        for(int colonne = 0 ; colonne < COLONNE_DAMIER; colonne++){
+            if(tab2[colonne][ligne]!=tab[colonne][ligne]){
+                return 1;
+            }
+        }
+    }
+    
+return 0;
+}
 
 
 //DEBUT PARTIE
 void jouer_dames(char tab[][COLONNE_DAMIER], char couleur){
 
-    char partie[20]= "start";
+    char tab2[COLONNE_DAMIER][LIGNE_DAMIER];
+
+    char partie= 'S';
     printf("Les BLANC commencent \n");
-    while(partie != "terminer")
-    {
+    while(partie != 'T'){
+        do{
+            for (int ligne = 0; ligne < LIGNE_DAMIER ; ligne++){
+                for(int colonne = 0 ; colonne < COLONNE_DAMIER; colonne++){
+                    tab2[colonne][ligne]=tab[colonne][ligne];
+                }
+            }
+            action_automatique(tab,couleur);
+
+            changement_dans_damier(tab,tab2);
```

Comment:
The call to changement_dans_damier(tab,tab2) is probably useless as the function doesn't change anything.